### PR TITLE
Upgrade react native to 0.71.3 (incremental upgrade 6)

### PR DESCRIPTION
**Pull request by samshara**
This PR is based on https://github.com/mapswipe/mapswipe/pull/750

Relevant Links:

- https://react-native-community.github.io/upgrade-helper/?from=0.70.7&to=0.71.3
- https://reactnative.dev/blog/2023/01/12/version-071
- https://github.com/facebook/react-native/blob/main/CHANGELOG.md#v071

- [x] Run `pod update && pod install`

### CODE REVIEW

**Comment by danbjoseph** [comment]
we had talked about trying to have smaller PRs with fewer files changed to make the review process easier. is this so large because it has wrapped up all the RN upgrades in one PR, and we didn't fully merge/release each version?

**Comment by udaynwa** [comment]
@danbjoseph , all the PRs have been individually tested. Since it would be the same to merge individually or merge all at once, we created one giant PR for this. The individual release versions are available in Releases and Testflight. Therefore, this is all ready to be reviewed and tested.

**Review by tnagorra — APPROVED** [review]

File: scripts/version.sh [hunk 1]
```diff
@@ -29,7 +29,7 @@ fi
 diff=`git diff-index HEAD | wc -l`
 if [[ $diff -gt 0 ]]; then
     echo "There are modified files in your working copy (or staged in the index). Please commit or stash them and rerun this command."
-    exit 1
+    # exit 1
```
Comment (laurentS) [inline scripts/version.sh:32]: Is this some debugging leftover? I'd really avoid leaving modified files in the tree while running this script.

File: scripts/version.sh [hunk 2]
```diff
@@ -80,7 +80,7 @@ then
 fi
 
 # run checks before creating the new version
-yarn lint
+# yarn lint
```
Comment (laurentS) [inline scripts/version.sh:83]: Any reason why the lint and flow steps are commented out? Debugging leftovers?

**Review by laurentS — APPROVED** [review]
Sorry for taking so long to review this. I am going with the assumption that you've tested it all under both android and iOS and all is good :)
Just left a couple of remarks about what looks like debugging leftovers. Otherwise, all good to me!

File: scripts/version.sh (hunk 1 above)
Comment (samshara) [reply scripts/version.sh]: @laurentS I have undo the changes and pushed.

File: scripts/version.sh (hunk 2 above)
Comment (samshara) [reply scripts/version.sh]: Seems so. I have made necessary changes.

**Comment by danbjoseph** [comment]
hi @laurentS, if you have no objection togglecorp will merge this into `dev` tomorrow so that work can continue on the new project off of the `dev` branch.

**Comment by laurentS** [comment]
All good from my end!

**Pull request merged**